### PR TITLE
Story collection 1 see

### PR DESCRIPTION
**Pull request by robings**
Story 1 ready for code review. The database name is "shinkansen"

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+require_once 'dbConnect.php';
+
+/**
+ * function to call a var dump with pre tags
+ * @param $variabletoDump - the variable to be dumped
+ */
+function var_dumpPre($variabletoDump) {
```
Comment (charliecog) [inline functions.php]: shouldn't be deployed so remove

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+require_once 'dbConnect.php';
```
Comment (charliecog) [inline functions.php]: require this in the index.php rather than here -

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * function to call a var dump with pre tags
+ * @param $variabletoDump - the variable to be dumped
```
Comment (charliecog) [inline functions.php]: line break to separate these

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ */
+function var_dumpPre($variabletoDump) {
+    echo '<pre>';
+    var_dump($variabletoDump);
+    echo '</pre>';
+}
+
+/**
+ * get the collection data from the db
+ * @param $db - the database connection
```
Comment (charliecog) [inline functions.php:6]: line breaks separating desc, param and return
(ill just comment this on this one - please repeat for all)

File: functions.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ * @return mixed - the data from the database in a assoc array
```
Comment (charliecog) [inline functions.php]: return is an array

File: functions.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+ */
+function fetchData($db) {
```
Comment (charliecog) [inline functions.php]: type hinting for the $db is PDO and the return type hinting is array

File: functions.php [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    } elseif (!array_key_exists('withdrawnYR', $shinkansen)) {
+        return 'error! missing expected array key(s): function collectionBox';
+    }
+    $shinkansenBox = '<article><div class=\'item\'>';
+    $shinkansenBox .= '<h2>Series ' . $shinkansen['series'] . '</h2>';
+    $shinkansenBox .= '<img src=\'' . $shinkansen['imgURL'] . '\' alt=\'Series ' . $shinkansen['series'] . ' Bullet Train\' />';
+    $shinkansenBox .= '<ul>';
+    $shinkansenBox .= '<li><span>Introduced:</span> ' . $shinkansen['introducedYR'] . '</li>';
+    $shinkansenBox .= '<li><span>Top speed:</span> ' . $shinkansen['topSpeedKMH'] . 'km/h (' . $shinkansen['topSpeedMPH'] . 'mph)</li>';
+    if ($shinkansen['withdrawn']==1) {
```
Comment (charliecog) [inline functions.php]: Could this if statement be a little neater as a ternary statement? would only need to target the content of the li

File: functions.php [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+ */
+function fetchData($db) {
+    $query = $db->prepare('SELECT `id`, `series`, `introducedYR`, `topSpeedKMH`, `topSpeedMPH`, `withdrawn`, `withdrawnYR`, `imgURL` FROM `shinkansens`;');
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**
+ * function to build the display box for a collection item
+ * @param array $shinkansen - array for one row from the db
```
Comment (charliecog) [inline functions.php]: doesnt have to be from the db, will handle any array with the correct keys. remove mention of the db

File: index.php [hunk 9]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+require_once 'functions.php';
+require_once 'dbConnect.php';
+
+$db = connectdb();
+
+
```
Comment (charliecog) [inline index.php:9]: remove extra line break

File: index.php [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+$shinkansens = fetchData($db);
```
Comment (charliecog) [inline index.php]: function name not semantic enough - getAllTrains makes more sense

File: index.php [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+$display = '';
+foreach($shinkansens as $shinkansen) {
```
Comment (charliecog) [inline index.php]: move the foreach to be inside the display function to increase the amount of testable code

File: index.php [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+?>
+
+<!DOCTYPE html>
+<html lang='en-GB'>
+<head>
+   <title>Shinkansen</title>
+    <meta name="viewport" content="width=device-width" />
+    <link rel='stylesheet' type='text/css' href='normalize.css' />
+    <link rel='stylesheet' type='text/css' href='styles.css' />
+    <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet">
```
Comment (charliecog) [inline index.php]: put this above where you use it in styles.css

File: index.php [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+</head>
+
+<body>
+<header>
+    <div class='heroImage'>
+        <h1>Shinkansen - Japanese Bullet Train Collection</h1>
+    </div>
+</header>
+<section class='collection'>
+    <!-- this section will be replaced by a php echo -->
```
Comment (charliecog) [inline index.php]: remove comment

File: index.php [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+    <?php
+        echo $display;
+    ?>
+
+    <!-- end of php echo section -->
```
Comment (charliecog) [inline index.php]: remove comment

File: index.php [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+<body>
+<header>
+    <div class='heroImage'>
+        <h1>Shinkansen - Japanese Bullet Train Collection</h1>
+    </div>
+</header>
+<section class='collection'>
+    <!-- this section will be replaced by a php echo -->
+    <?php
+        echo $display;
```
Comment (charliecog) [inline index.php]: do this all in one line

Comment (charliecog) [reply index.php]: and name the variable more semantically eg: $trains

File: index.php [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+    ?>
+
+    <!-- end of php echo section -->
+</section>
+
+</body>
+
+</html>
+
+<?php
```
Comment (charliecog) [inline index.php]: remove all of this stuff below

File: tests/functions.php [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+        $case= collectionBox($input);
+        $this->assertEquals($expected, $case);
+    }
+    public function testCollectionBoxFailure() {
+        $expected = 'error! missing expected array key(s): function collectionBox';
+        $input = [ 'id' =>'1', 'version'=>'0', 'introducedYR'=>'1964', 'topSpeedKMPH'=>'220', 'topSpeedMPH'=>'135', 'withdrawn'=>'1', 'withdrawnYR'=>'2008', 'imgURL'=>'images/japan-974730_1920.jpg'];
+        $case= collectionBox($input);
+        $this->assertEquals($expected, $case);
+    }
+    public function testCollectionBoxFailure2() {
```
Comment (charliecog) [inline tests/functions.php]: make the name of the test more descriptive eg: testCollectionBoxFailureMissingIdKey

File: index.php [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+require_once 'dbConnect.php';
+
+$db = connectdb();
+
+
+$shinkansens = fetchData($db);
+
+$display = '';
+foreach($shinkansens as $shinkansen) {
+    $display .= collectionBox($shinkansen);
```
Comment (charliecog) [inline index.php]: collection box name not the most semantic - maybe something like displayTrains()

**Review by charliecog — COMMENTED** [review]
Looks good - almost there! nice build so far!

**Comment by robings** [comment]
Hello, I've made the changes

**Review by charliecog — APPROVED** [review]

**Pull request merged**